### PR TITLE
Add LOS model

### PR DESCRIPTION
**Pull request by Andrew-Dickinson**
Adds a model to represent the concept of a "line of sight" (LOS) between two building objects. The presence of an object in this new table indicates the pair of indicated buildings is able to see each other, as of a provided date.

Primarily right now, we use this to encode "potential" links, since the `Link` model can't really encode this very well (see #402). However, in the future, this table could be used to store results from automated LOS analysis tools as well as site surveys and Google earth analysis, etc.

At import time, we create LOS objects for any existing links in the spreadsheet or UISP. At map and KML render time, we convert back to "planned" link objects so that they are appropriately displayed

### CODE REVIEW

File: src/meshapi/views/geography.py [hunk 1]
```diff
@@ -210,67 +219,127 @@ def get(self, request: HttpRequest) -> HttpResponse:
                 folder.append(node_placemark)
                 mapped_nns.add(install.node.network_number)
 
+        all_links_set = set()
+        kml_links: List[LinkKMLDict] = []
         for link in (
             Link.objects.prefetch_related("from_device")
             .prefetch_related("to_device")
             .filter(~Q(status=Link.LinkStatus.INACTIVE))
+            .exclude(type=Link.LinkType.VPN)
```
Comment (Andrew-Dickinson) [inline src/meshapi/views/geography.py:228]: This is not related to the LOS stuff, I just wanted to get rid of these

File: src/meshapi/views/map.py [hunk 2]
```diff
@@ -213,6 +214,79 @@ def list(self, request: Request, *args: List[Any], **kwargs: Dict[str, Any]) ->
...
+                            & Q(to_device__node__buildings=OuterRef("to_building"))
+                        )
+                        | (
+                            Q(from_device__node__buildings=OuterRef("to_building"))
+                            & Q(to_device__node__buildings=OuterRef("from_building"))
+                        )
+                    )
+                )
+            )
+            .filter(
```
Comment (WillNilges) [inline src/meshapi/views/map.py:244]: "It works™, we have Tests ;)"
- Andrew, 2024

File: src/meshdb/utils/spreadsheet_import/fetch_uisp.py [hunk 3]
```diff
@@ -29,6 +30,18 @@ def download_uisp_links():
     )
 
 
+def get_uisp_device_detail(device_id: str, session: Optional[requests.Session] = None) -> dict:
```
Comment (WillNilges) [inline src/meshdb/utils/spreadsheet_import/fetch_uisp.py:33]: Add a comment about the kind of data we get or maybe elaborate the `dict`?

**Review by WillNilges — APPROVED** [review]

File: src/meshapi/views/geography.py [hunk 4]
```diff
@@ -210,67 +219,127 @@ def get(self, request: HttpRequest) -> HttpResponse:
...
+                            los.from_building.altitude or DEFAULT_ALTITUDE,
+                        ),
+                        "to_coord": (
+                            los.to_building.longitude,
+                            los.to_building.latitude,
+                            los.to_building.altitude or DEFAULT_ALTITUDE,
+                        ),
+                        "extended_data": {
+                            "name": f"LOS-{los.id} {link_label}",
+                            "stroke": INACTIVE_COLOR,
```
[the remark this answers is not in the document]
Comment (Andrew-Dickinson) [reply src/meshapi/views/geography.py]: Done

File: src/meshapi/views/map.py (hunk 2 above)
Comment (Andrew-Dickinson) [reply src/meshapi/views/map.py:244]: I stand by this

File: src/meshdb/utils/spreadsheet_import/fetch_uisp.py (hunk 3 above)
Comment (WillNilges) [reply src/meshdb/utils/spreadsheet_import/fetch_uisp.py:33]: "meh"
- Andrew, 2024

Comment (Andrew-Dickinson) [reply src/meshdb/utils/spreadsheet_import/fetch_uisp.py:33]: Nah UISP sucks
